### PR TITLE
* #100: Added release priority: 1. MAVEN, 2.GITHUB.

### LINKED ISSUE #97: Improve logging when the version validation fails
currently if the new version to be released is invalid, the display error is `E-RR-VAL-4: A new version does not fit the versioning rules. Possible versions for the release are: [0.2.0, 0.1.1, 1.0.0]`

It would help to see what is actually the actual wrong version here as well.

### LINKED ISSUE #100: Add a built-in release priority map
## Problem

Current releases available: GITHUB, MAVEN

Priority:
1. MAVEN
2. GITHUB

If the Maven release fails, it makes no sense to run the GitHub release.

We don't want to rely on the order of releases that come from user input.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #100 
Closes #97

### CODE REVIEW

File: src/main/java/com/exasol/releasedroid/usecases/validate/GitRepositoryValidator.java [hunk 1]
```diff
@@ -73,7 +73,8 @@ private Report validateNewVersionWithPreviousTag(final String newTag, final Stri
             report.addResult(ValidationResult.successfulValidation("A new tag."));
         } else {
             report.addResult(ValidationResult.failedValidation("E-RR-VAL-4",
-                    "A new version does not fit the versioning rules. Possible versions for the release are: "
+                    "A new version '" + newTag
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/validate/GitRepositoryValidator.java:76]:
```suggestion
                    "The new version '" + newTag
```

File: src/test/java/com/exasol/releasedroid/usecases/validate/GitRepositoryValidatorTest.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ void testValidateVersionWithPreviousTagInvalid(final String version) {
         final Report validationReport = this.validator.validateNewVersion(version, this.gitRepositoryMock);
         assertAll(() -> assertThat(validationReport.hasFailures(), equalTo(true)),
                 () -> assertThat(validationReport.getFailuresReport(),
-                        containsString("E-RR-VAL-4: " + "A new version does not fit the versioning rules. "
+                        containsString("E-RR-VAL-4: A new version '" + version + "' does not fit the versioning rules. "
```
Comment (chiaradiamarcelo) [inline src/test/java/com/exasol/releasedroid/usecases/validate/GitRepositoryValidatorTest.java:153]:
```suggestion
                        containsString("E-RR-VAL-4: The new version '" + version + "' does not fit the versioning rules. "
```

File: src/main/java/com/exasol/releasedroid/usecases/release/ReleaseInteractor.java [hunk 3]
```diff
@@ -52,13 +56,17 @@ public ReleaseInteractor(final ValidateUseCase validateUseCase,
...
         final Report report = ReportImpl.releaseReport();
         final Repository repository = this.repositoryGateway.getRepositoryWithDefaultBranch(repositoryFullName);
-        for (final PlatformName platformName : platformNames) {
-            try {
-                this.getReleaseMaker(platformName).makeRelease(repository);
-                report.addResult(ReleaseResult.successfulRelease(platformName));
-            } catch (final ReleaseException exception) {
-                report.addResult(ReleaseResult.failedRelease(platformName, ExceptionUtils.getStackTrace(exception)));
-                break;
+        for (final PlatformName platformName : this.releasePriority) {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/release/ReleaseInteractor.java:59]: it's a little bit tricky to understand the whole flow. I suggest to extract this to a dedicated method, something like `List<PlatformName> orderByPriority(platformNames)` or similar

File: src/main/java/com/exasol/releasedroid/usecases/release/ReleaseInteractor.java [hunk 4]
```diff
@@ -18,6 +21,7 @@
     private final ValidateUseCase validateUseCase;
     private final Map<PlatformName, ? extends ReleaseMaker> releaseMakers;
     private final RepositoryGateway repositoryGateway;
+    private final List<PlatformName> releasePriority = List.of(MAVEN, GITHUB);
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/release/ReleaseInteractor.java:24]: this should be moved to the constructor. Remember the business rules have no idea about `Maven` or `Github`

File: src/main/java/com/exasol/releasedroid/usecases/PlatformName.java [hunk 5]
```diff
@@ -11,16 +11,17 @@
...
+     * @return list of members of {@link PlatformName} enum class
      */
-    public static Set<PlatformName> toSet(final String... platforms) {
-        final Set<PlatformName> platformsList = new HashSet<>();
+    public static List<PlatformName> toList(final String... platforms) {
+        final List<PlatformName> platformsList = new ArrayList<>();
         for (final String platform : platforms) {
             platformsList.add(getPlatformName(platform));
         }
+        platformsList.sort(Comparator.comparingInt(List.of(MAVEN, GITHUB)::indexOf));
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/PlatformName.java:24]: if we are moving this, please anyway extract it to a dedicated method to know why we are doing it.

**Review by chiaradiamarcelo — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/releasedroid/usecases/PlatformName.java (hunk 5 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releasedroid/usecases/PlatformName.java]: Fixed it

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**